### ISSUE TITLE
example using tail command

### ISSUE THREAD

**Issue by ORESoftware** [junosuarez/exec-stream#4]
Hi there

I am trying to read the last few lines of a large file on a Mac/UNIX filesystem using node.js.

from the command line, this works:

`tail -n 5 /file/path` (reads/writes the last 5 lines of a file)

so now I am trying to run that command from inside node.js

but this isn't working

``` javascript
process.stdin
        .pipe(execStream('tail -n 3 ' + stderr_path))
        .pipe(process.stdout);
```

I get an ENOENT error, so something is wrong when attempting to open the file

so I started making sh*t up, throwing the sudo word around like so

``` javascript
process.stdin
        .pipe(execStream('tail -n 3 ' + stderr_path,['sudo']))
        .pipe(process.stdout);
```

I even tried starting my server with sudo, like  "sudo NODE_ENV=xxx node start.js"

no dice

any ideas on how to get this work? Does execStream always have to be an argument to process.stdin.pipe()? If not can you show an example that doesn't use that? thx

**Comment by ORESoftware** [comment]
I was able to get this to work

``` javascript
var cp = require('child_process');

cp.exec("tail -n 5",function(err,stdout,stderr){

})
```

the problem with that is that it requires a child_process which can be expensive. Is there a way to get the same functionality with this lib?

**Comment by junosuarez** [comment]
> Does execStream always have to be an argument to process.stdin.pipe()

No. `execStream` returns a duplex stream, meaning you can (optionally) pipe any stream into it as standard in, which would be the equivalent of piping something into it in bash, for example `cat readme.md | node example.js`

>  If not can you show an example that doesn't use that?

As a trivial example, the following two programs behave the same:

sh script:

``` sh
echo hello
```

javascript:

``` js
var execStream = require('exec-stream')
execStream('echo hello').pipe(process.stdout)
```

> it requires a child_process which can be expensive. Is there a way to get the same functionality with this lib?

This library uses child_process.spawn
